pkg/services/ngalert/schedule/schedule.go

```diff
@@ -530,6 +563,22 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key models.AlertRul
 	defer sch.stopApplied(key)
 	for {
 		select {
+		// used by external services (API) to notify that rule is updated.
+		case <-updateCh:
```

Comment:
I am thinking if we can make it more apparent that both `<-updateCh` and `<-evalCh` have the same prelude (i.e. both of them call `updateRule`, write the same log lines with the same message, and update `currentRule`. I think we could even remove `updateRule` from `<-evalCh` if we check `<-updateCh` first?

Reply:
I thought about that too but then could find a use case when this is not true - Grafana HA mode.